app/components/custom_status/custom_status_expiry.tsx

```diff
@@ -0,0 +1,157 @@
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {getCurrentMomentForTimezone} from '@utils/timezone';
+
+import {Theme} from '@mm-redux/types/preferences';
+import {GlobalState} from '@mm-redux/types/store';
+
+type Props = {
+    theme: Theme;
+    time: Date;
+    textStyles?: TextStyle;
+    testID?: string;
+    showPrefix?: boolean;
+    withinBrackets?: boolean;
+    showToday?: boolean;
+    showTimeCompulsory?: boolean;
+}
+
+const CustomStatusExpiry = ({time, theme, textStyles = {}, showPrefix, withinBrackets, testID = '', showTimeCompulsory, showToday}: Props) => {
+    const timezone = useSelector(getCurrentUserTimezone);
+    const styles = createStyleSheet(theme);
+    const militaryTime = useSelector((state: GlobalState) => getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, 'use_military_time'));
+    const currentMomentTime = getCurrentMomentForTimezone(timezone);
+
+    const expiryMomentTime = timezone ? moment(time).tz(timezone) : moment(time);
+
+    const plusSixDaysEndTime = currentMomentTime.clone().add(6, 'days').endOf('day');
+    const tomorrowEndTime = currentMomentTime.clone().add(1, 'day').endOf('day');
+    const todayEndTime = currentMomentTime.clone().endOf('day');
+    const isCurrentYear = currentMomentTime.get('y') === expiryMomentTime.get('y');
+
+    let format = '';
+    let renderDayOrDate;
+
+    if (showToday && expiryMomentTime.isSameOrBefore(todayEndTime)) {
+        renderDayOrDate = (
+            <Text>
+                <FormattedText
+                    id='custom_status.expiry_time.today'
+                    defaultMessage='Today'
+                />
+                {' '}
+                <FormattedText
+                    id='custom_status.expiry.at'
+                    defaultMessage='at'
+                />
+                {' '}
+            </Text>
+        );
+    } else if (expiryMomentTime.isSame(todayEndTime)) {
+        renderDayOrDate = (
+            <FormattedText
+                id='custom_status.expiry_time.today'
+                defaultMessage='Today'
+            />
+        );
+    } else if (expiryMomentTime.isAfter(todayEndTime) && expiryMomentTime.isSameOrBefore(tomorrowEndTime)) {
+        renderDayOrDate = (
+            <Text>
+                <FormattedText
+                    id='custom_status.expiry_time.tomorrow'
+                    defaultMessage='Tomorrow'
+                />
+                {' '}
+                <FormattedText
+                    id='custom_status.expiry.at'
+                    defaultMessage='at'
+                />
+                {' '}
+            </Text>
+        );
+    }
+
+    const useDay = expiryMomentTime.isBetween(tomorrowEndTime, plusSixDaysEndTime);
+    const useDate = expiryMomentTime.isAfter(plusSixDaysEndTime) && isCurrentYear;
+
+    if (useDay) {
+        format = 'dddd';
+    } else if (useDate) {
+        format = 'MMM DD';
+    } else if (!isCurrentYear) {
+        format = 'MMM DD, YYYY';
+    }
```

Comment:
creating the variables `useDay` and `useDate` are not necessary

```suggestion
    if (expiryMomentTime.isBetween(tomorrowEndTime, plusSixDaysEndTime)) {
        format = 'dddd';
    } else if (expiryMomentTime.isAfter(plusSixDaysEndTime) && isCurrentYear) {
        format = 'MMM DD';
    } else if (!isCurrentYear) {
        format = 'MMM DD, YYYY';
    }
```